File: Solved/Reshte Ramzi/ReshteRamzi.java
import java.util.Scanner;

/*
Question title = reshte ramzi - رشته رمزی
https://quera.ir/problemset/106796/
 */

public class ReshteRamzi {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String length = sc.nextLine();
        int count = Integer.parseInt(sc.nextLine());
        String text = sc.nextLine();

        for (int i = 0; i <= count - 1; i++) {
            String a = lastToFirst(text);
            text = replace(a);
        }
        System.out.println(text);
    }

    //move last letter to first
    private static String lastToFirst(String text) {
        char l = text.charAt(text.length() - 1);
        StringBuilder builder = new StringBuilder(text);
        builder.deleteCharAt(text.length() - 1);
        return l + builder.toString();
    }

    //replace all letters with their next letter in alphabet
    private static String replace(String text) {
        StringBuilder result = new StringBuilder();
        for (char i : text.toCharArray()) {
            if (i == 'z') {
                result.append('a');
            } else {
                result.append((char) (((int) i) + 1));
            }
        }
        return result.toString();
    }
}

File: Solved/Reshte Ramzi/Reshte Ramzi.py
# رشته‌ی رمزی
# ID : 106796
# https://quera.org/problemset/106796/
# https://b2n.ir/p38675

def lastToFirst(text):
    return text[-1] + text[:-1]


def replacer(text):
    result = []

    for i in text:
        if (i == 'z'):
            result.append('a')

        else:
            result.append(chr((ord(i) + 1)))

    return "".join(result)


length = input()
count = int(input())
text = input()

for i in range(count):
    a = lastToFirst(text)
    text = replacer(a)

print(text)
